src/mainwindow.cc

```diff
@@ -1170,3 +1200,139 @@ void MainWindow::onShowVolumeMetersCheckButtonToggled(bool toggled) {
+    connect(&systrayIcon, SIGNAL(activated(QSystemTrayIcon::ActivationReason)),
+                this, SLOT(iconActivated(QSystemTrayIcon::ActivationReason)));
+    connect(enableSystrayCheckButton, SIGNAL(clicked()), this, SLOT(toggleSystrayOption()));
+
+    systrayIconMenu.addAction(&systrayMinimizeAction);
+    systrayIconMenu.addAction(&systrayRestoreAction);
+    systrayIconMenu.addSeparator();
+    systrayIconMenu.addAction(&systrayQuitAction);
+
+    QSize sz(256,256);
+    systrayIcons[NOT_MUTED].addPixmap(style()->standardIcon(QStyle::SP_MediaVolume).pixmap(sz));
+    systrayIcons[MUTED].addPixmap(style()->standardIcon(QStyle::SP_MediaVolumeMuted).pixmap(sz));
+
+    systrayIcon.setIcon(systrayIcons[NOT_MUTED]);
+    systrayIcon.setContextMenu(&systrayIconMenu);
+    if(systrayEnabled())
+        systrayIcon.show();
+}
+
+void MainWindow::toggleSystrayOption(){
+    if(systrayIcon.isVisible()){
+        systrayIcon.hide();
+        startInSystrayCheckButton->setDisabled(true);
+        closeToSystrayCheckButton->setDisabled(true);
+    } else {
+        startInSystrayCheckButton->setDisabled(false);
+        closeToSystrayCheckButton->setDisabled(false);
+        systrayIcon.show();
+    }
+}
+
+void MainWindow::systrayMuteToggle()
+{
+    static bool muted = false;
+    static std::map<uint32_t, bool> restore_mute_state;
+    
+    for (std::map<uint32_t, SinkWidget*>::iterator it = sinkWidgets.begin(); it != sinkWidgets.end(); ++it) {
+        if(!muted) {
+            restore_mute_state[it->first] =  it->second->muteToggleButton->isChecked();
+            it->second->muteToggleButton->setChecked(true);
+            systrayIcon.setIcon(systrayIcons[MUTED]);
+        } else {
+            if(restore_mute_state.find(it->first) != restore_mute_state.end())
+                it->second->muteToggleButton->setChecked(restore_mute_state[it->first]);
+            else
+                continue; // skip unnecessary state update, we aren't restoring anything
+            systrayIcon.setIcon(systrayIcons[NOT_MUTED]);
+        }
+        it->second->onMuteToggleButton();
+    }
+    muted = !muted;
+}
+
+void MainWindow::systrayVolumeChange(int step)
+{
+    for (std::map<uint32_t, SinkWidget*>::iterator it = sinkWidgets.begin(); it != sinkWidgets.end(); ++it) {
+        DeviceWidget* w = dynamic_cast<DeviceWidget*>(it->second);
+        if(!w || !w->channels | !w->channels[0])
+                continue;
+        Channel *c = w->channels[0];
+        QSlider *vs = c->volumeScale;
+        int vol = vs->value();
+        int max = vs->maximum();
+        int min = vs->minimum();
+        double single_percent_step = (double)(max-min) / 100;
+        vs->setValue(vol + step*single_percent_step);
+    }
+}
+
+void MainWindow::iconActivated(QSystemTrayIcon::ActivationReason reason){
+    switch(reason)
+    {
+    case QSystemTrayIcon::Trigger:
+    case QSystemTrayIcon::DoubleClick:
```

Comment:
Just a matter of taste.... but I would suggest left-click -> show/hide window, middle-click-> mute/unmute.

Reply:
Makes sense.